dao/src/main/scala/za/co/absa/enceladus/dao/rest/RestClient.scala

```diff
@@ -76,23 +83,37 @@ protected class RestClient(authClient: AuthClient,
 
     val statusCode = response.getStatusCode
 
+    /**
+     * This function handles unauthorized response by trying to authenticate
+     * (if there are still some retries attempt left) - this might be due to an expired session.
+     *
+     * @param exceptionToThrow Instance of an exception that should be thrown if there are no retries left.
+     */
+    def handleUnauthorizedResponse(exceptionToThrow: OptionallyRetryableException): Unit = {
+      log.warn(s"Response - ${statusCode} : ${Option(response.getBody).getOrElse("None")}")
```

Comment:
```suggestion
      log.warn(s"Response - ${statusCode} : ${Option(response.getBody).getOrElse("<empty body>")}")
```
or just 
```suggestion
      log.warn(s"Response - ${statusCode} : ${Option(response.getBody).getOrElse("")}")
```
or something similar - `None` makes little sense when expecting a body from the log-reader's perspective.

Reply:
I like it :)